starfish/util/exec.py

```diff
@@ -36,11 +35,17 @@ def stages(commands: Sequence[Sequence[Union[str, Callable]]],
 
     Environment variables
     ---------------------
-    STARFISH_COVERAGE:
+    STARFISH_COVERAGE :
          If set, then command lists will have `coverage run ...` options
          prepended before execution.
     """
-    tempdir = tempfile.mkdtemp()
+
+    if keep_data:
```

Comment:
I could have sworn there was a `delete` option for `tempfile.TemporaryDirectory()`, but clearly I'm wrong.  In that case, I think the original code is ok as `tempfile.TemporaryDirectory()` doesn't really provide any significant improvement.

Reply:
Happy to rollback, but the use of TD didn't feel particularly dirtier than of shutil.

Reply:
Ok, I'll leave it up to your judgement.